### PR TITLE
TS-5025: Add 'setHost' method to 'atscppapi::Request'

### PR DESCRIPTION
**Pull request by SolidWallOfCode**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1134/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1028/ for details.

File: lib/cppapi/Request.cc [hunk 1]
```diff
@@ -177,6 +177,23 @@ Request::getHeaders() const
   return state_->headers_;
 }
 
+void
+Request::setHost(std::string const &host)
+{
+  static std::string HOST_FIELD_NAME(TS_MIME_FIELD_HOST, TS_MIME_LEN_HOST);
```
Comment (zwoop) [inline lib/cppapi/Request.cc]: < bike_shed >Shouldn't this be const too?< /bike_shed >

Comment (SolidWallOfCode) [inline lib/cppapi/Request.cc]: Yes it should.

**Comment by SolidWallOfCode** [comment]
Yeah, we had a problem at Y! this fixed. The plugin writer set the `HOST` header in order to control subsequent remapping, a reasonable thing to do. But if the request comes in with the host in the URL then remap prioritizes that over the `HOST` header and things don't work right for non obvious reasons. Better to have a real "set the host" method that does the right thing. Technically this could be done by the plugin but the point of the CPP API is to make writing plugins easier by taking care of these pesky details that would be the same in every plugin.

**Pull request merged**